ls1intum/Artemis
# `Adaptive learning`: Improve learning path graph view

### Checklist
#### General


#### Server
#### Client

### Motivation and Context



This PR improves usability and delivers a more dynamic UI for the learning path graph view.

### Description

- nodes are now larger
- competency start nodes now display the competency progress cycles
- the view contains a mini-map to help the user navigate
- when clicking on competencies users get redirected to the corresponding competency card

### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 Student
- 1 Course with LPs enabled, multiple competencies and relations

1. Log in to Artemis
2. Navigate to competency management
3. open the graph view of a student's learning path
4. zoom in and out and play with the mini-map, make sure it works as expected
5. make sure the corresponding competency rings are displayed for competency start nodes
6. Click on a competency node (start & end node) and check if you are redirected to the correct competency card

### Review Progress






#### Performance Review
#### Code Review
#### Manual Tests

### Test Coverage







### Screenshots



[Screencast](https://youtu.be/QkuQkIREoYk)

Screenshot:
![chrome_KszM9wQY54](https://github.com/ls1intum/Artemis/assets/44003963/74e43f10-134f-4bba-b401-9ebd1f32491f)

### Testserver States
> [!NOTE]
> These badges show the state of the test servers.
> Green = Currently available, Red = Currently locked

![](https://byob.yarr.is/ls1intum/Artemis/artemis-test1)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test2)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test3)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test4)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test5)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test6)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test9)
